libs/components/packages/src/schematics/migrations/update-8/update-polyfill/update-polyfill.schematic.ts

```diff
@@ -0,0 +1,61 @@
+import { Rule, chain } from '@angular-devkit/schematics';
+import * as ts from '@schematics/angular/third_party/github.com/Microsoft/TypeScript/lib/typescript';
+
+import { addPolyfillsConfig } from '../../../rules/add-polyfills-config';
+import { getWorkspace } from '../../../utility/workspace';
+
+function removePolyfillCode(): Rule {
+  return async (tree): Promise<void> => {
+    const { workspace } = await getWorkspace(tree);
+    workspace.projects.forEach((project) => {
+      ['polyfills.ts', 'test.ts'].forEach((filename) => {
+        const path = `${project.root}/src/${filename}`;
+        if (tree.exists(path)) {
+          const polyfillsFile = tree.readText(path);
+          const sourceFile = ts.createSourceFile(
+            path,
+            polyfillsFile,
+            ts.ScriptTarget.Latest,
+            true
+          );
+          // Find the expression that assigns the global property to the window object.
+          const expression = sourceFile.statements.find((node) => {
+            if (ts.isExpressionStatement(node)) {
+              const expression = node.expression;
+              if (ts.isBinaryExpression(expression)) {
+                const left = expression.left;
+                if (ts.isPropertyAccessExpression(left)) {
+                  let leftExpression: ts.Expression = left.expression;
+                  if (ts.isParenthesizedExpression(leftExpression)) {
+                    leftExpression = leftExpression.expression;
+                  }
+                  if (
+                    ts.isAsExpression(leftExpression) &&
+                    ts.isIdentifier(leftExpression.expression) &&
+                    leftExpression.expression.text === 'window' &&
+                    left.name.text === 'global'
+                  ) {
+                    const right = expression.right;
+                    if (ts.isIdentifier(right) && right.text === 'window') {
+                      return true;
+                    }
+                  }
+                }
+              }
+            }
+            return false;
+          });
+          if (expression) {
+            const change = tree.beginUpdate(path);
+            change.remove(expression.pos, expression.end - expression.pos);
```

Comment:
Does this also remove the comment we added to the `test.ts` file?
```
// Fix for crossvent "global is not defined" error. The crossvent library
// is used by Dragula, which in turn is used by multiple SKY UX components.
// See: https://github.com/bevacqua/dragula/issues/602
(window as any).global = window;
```

Reply:
This is what we add to the `polyfills.ts` file:
```
/***************************************************************************************************
 * SKY UX POLYFILLS - DO NOT MODIFY THIS SECTION
 */

// Fix for crossvent `global is not defined` error. The crossvent library is used by Dragula,
// which in turn is used by multiple SKY UX components.
// https://github.com/bevacqua/dragula/issues/602
(window as any).global = window;

/*
 * END SKY UX POLYFILLS
 **************************************************************************************************/
```

Reply:
The AST includes the preceeding comment in the expression. I've added a special case to look for our comment as it was removing the opening comment but not the `END` comment.